### PR TITLE
Allow HTTP Headers in Tracer

### PR DESCRIPTION
**Pull request by magikid**
This adds support for HTTP Headers when extracting and injecting the
context.  To use, the carrier needs to implement the
`RequestInterface` interface.

Issue: #22

### CODE REVIEW

**Comment by jcchavezs** [comment]
Hi @magikid thanks for pointing our this. Could you cover https://github.com/jcchavezs/zipkin-php-opentracing/pull/23/files#diff-25c66202aad94dec22819d4ca7e347b2R74 with a test?

**Comment by magikid** [comment]
I don't know why I didn't when I first submitted this.  All I can say was that it was Friday 😄

**Comment by jcchavezs** [comment]
Not sure why travis is failing. Investigating it.

**Comment by magikid** [comment]
I see why. Style violations from my code. I'll update the PR in a few hours.

**Comment by jcchavezs** [comment]
@magikid have you been able to sort this out?

**Comment by magikid** [comment]
Well it took me significantly longer than a few hours to resolve those style issues but the checks are all passing now.

**Comment by jcchavezs** [comment]
Great work @magikid, do you have any input @ellisv (or @eligijusvitkauskas-home24)?

File: tests/Unit/HeaderClass.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace ZipkinOpenTracing\Tests\Unit;
+
+use function array_keys;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\StreamInterface;
+use Psr\Http\Message\UriInterface;
+
+class HeaderClass implements RequestInterface
```
Comment (jcchavezs) [inline tests/Unit/HeaderClass.php]: nitpick, do you mind making this class final?

File: tests/Unit/HeaderClass.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+{
+    /** @var array $headers */
```
Comment (jcchavezs) [inline tests/Unit/HeaderClass.php]: Could we please do 
```
/**
 *@var array $headers
 */
```

instead?

File: src/ZipkinOpenTracing/Tracer.php [hunk 3]
```diff
@@ -70,6 +71,11 @@ public function startActiveSpan($operationName, $options = [])
             $options = StartSpanOptions::create($options);
         }
 
+        if (!$this->hasParentInOptions($options) && $this->getActiveSpan() !== null) {
```
Comment (jcchavezs) [inline src/ZipkinOpenTracing/Tracer.php:74]: Could we please reflect this piece of logic in the unit test?

**Comment by jcchavezs** [comment]
Thanks for this @magikid, I left some comments, once they are fixed I am happy to merge this.

**Comment by jcchavezs** [comment]
Merged with #29. Thanks a lot @magikid!

**Pull request closed without merging**